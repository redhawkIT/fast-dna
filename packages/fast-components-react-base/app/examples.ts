import ButtonExamples from "../src/button/examples.data";
export { ButtonExamples };
