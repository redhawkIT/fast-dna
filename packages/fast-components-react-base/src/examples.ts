import * as React from "react";

export type IExampleData<T> = T;

export interface IGenericExample<T> {
    name: string;

    component: React.ComponentClass;

    data: Array<IExampleData<T>>;
}
