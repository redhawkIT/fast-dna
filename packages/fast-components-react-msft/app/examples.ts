import ButtonExample from "../src/button/examples.data";
export {ButtonExample};

import DividerExample from "../src/divider/examples.data";
export {DividerExample};

import ToggleExamples from "../src/toggle/examples.data";
export {ToggleExamples};
