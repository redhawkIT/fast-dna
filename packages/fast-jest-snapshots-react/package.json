{
  "name": "@microsoft/fast-jest-snapshots-react",
  "description": "A small utility for automating Jest snapshot creation of React components",
  "version": "1.6.0",
  "author": {
    "name": "Microsoft",
    "email": "fwq@microsoft.com"
  },
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Microsoft/fast-dna.git"
  },
  "private": true,
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "scripts": {
    "prepare": "tsc -p ./tsconfig.json",
    "test": "npm run tslint",
    "start": "webpack-dev-server",
    "tslint": "tslint -c ./tslint.json '**/*.ts{,x}'",
    "tslint:fix": "tslint -c ./tslint.json --fix '**/*.ts{,x}'"
  },
  "devDependencies": {
    "@microsoft/fast-tslint-rules": "^1.6.0",
    "@types/jest": "^22.2.2",
    "jest": "^22.4.3",
    "react": "^16.3.0",
    "ts-jest": "^22.4.2",
    "tslint": "^5.9.1",
    "typescript": "^2.8.1"
  },
  "dependencies": {
    "@types/react": "^16.3.0",
    "react-test-renderer": "^16.3.0"
  },
  "peerDependencies": {
    "jest": "^22.0.0",
    "react": "^16.3.0"
  }
}
