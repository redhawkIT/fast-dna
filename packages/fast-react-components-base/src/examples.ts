import * as React from "react";

export type IExampleData<T> = T;

export interface IGenericExample<T> {
    name: string;

    component: React.ComponentClass;

    data: IExampleData<T>[];
}