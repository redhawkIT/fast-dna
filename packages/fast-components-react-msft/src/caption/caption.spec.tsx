import * as React from "react";
import * as Adapter from "enzyme-adapter-react-16";
import { configure, shallow } from "enzyme";
import examples from "./examples.data";
import { generateSnapshots } from "@microsoft/fast-jest-snapshots-react";
import MSFTCaption, {
    CaptionProps,
    CaptionSize,
    CaptionTag,
    ICaptionHandledProps,
    ICaptionManagedClasses,
    ICaptionUnhandledProps
} from "./caption";
import { Caption } from "./index";

/**
 * Configure Enzyme
 */
configure({adapter: new Adapter()});

describe("caption snapshots", (): void => {
    generateSnapshots(examples);
});

describe("caption", (): void => {
    test("should have a displayName that matches the component name", () => {
        expect((MSFTCaption as any).name).toBe(MSFTCaption.displayName);
    });

    test("should not throw if managedClasses are not provided", () => {
        expect(
            () => {
                shallow(<MSFTCaption />);
            }
        ).not.toThrow();
    });

    test("should return an object that includes all valid props which are not enumerated as handledProps", () => {
        const handledProps: ICaptionHandledProps = {
            tag: CaptionTag.p,
            size: CaptionSize._1
        };

        const unhandledProps: ICaptionUnhandledProps = {
            "aria-hidden": true
        };

        const props: ICaptionHandledProps & ICaptionUnhandledProps = {...handledProps, ...unhandledProps};

        const rendered: any = shallow(
            <Caption {...props} />
        );

        const caption: any = rendered.first().shallow();

        expect(caption.prop("aria-hidden")).toEqual(true);
    });

    test("should render a default `tag` of `CaptionTag.p` if no `tag` prop is passed", () => {
        const rendered: any = shallow(
            <Caption />
        );
        const caption: any = rendered.first().shallow();

        expect(caption.instance().props.tag).toEqual(CaptionTag.p);
    });

    test("should render the correct `size` when `size` prop is passed", () => {
        const rendered: any = shallow(
            <Caption size={CaptionSize._2} />
        );
        const caption: any = rendered.first().shallow();

        expect(caption.instance().props.size).toEqual(CaptionSize._2);
    });
});
