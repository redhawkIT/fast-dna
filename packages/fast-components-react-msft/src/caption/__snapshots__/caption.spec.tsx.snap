// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`caption snapshots Caption: 0 1`] = `
<p
  className="typography_8-0-1-11 caption-0-1-1 caption_1-0-1-2"
>
  Caption test string default
</p>
`;

exports[`caption snapshots Caption: 1 1`] = `
<p
  className="typography_8-0-1-23 caption-0-1-13 caption_1-0-1-14"
>
  Caption test string level 1
</p>
`;

exports[`caption snapshots Caption: 2 1`] = `
<p
  className="typography_9-0-1-36 caption-0-1-25 caption_2-0-1-27"
>
  Caption test string level 2
</p>
`;

exports[`caption snapshots Caption: 3 1`] = `
<span
  className="typography_8-0-1-47 caption-0-1-37 caption_1-0-1-38"
>
  span tag test string
</span>
`;

exports[`caption snapshots Caption: 4 1`] = `
<caption
  className="typography_8-0-1-59 caption-0-1-49 caption_1-0-1-50"
>
  caption tag test string
</caption>
`;

exports[`caption snapshots Caption: 5 1`] = `
<figcaption
  className="typography_8-0-1-71 caption-0-1-61 caption_1-0-1-62"
>
  figcaption tag test string
</figcaption>
`;
